File: benchmarks/suite5/powmod.go
package main


func main() {
	acc := int64(1)
	const base int64 = 3
	n := suite5OpaqueI64(2500000)
	for i := int64(0); i < n; i++ {
		acc = (acc * base) % 1000000007
	}
	suite5PrintI64(acc)
}

File: benchmarks/suite5/powmod.rs
mod bench_runtime;
use bench_runtime::{suite5_opaque_i64, suite5_print_i64};

fn main() {
    let mut acc: i64 = 1;
    let base: i64 = 3;
    let n: i64 = suite5_opaque_i64(2500000);
    for _ in 0..n {
        acc = (acc * base) % 1_000_000_007;
    }
    suite5_print_i64(acc);
}
